src/hints/patricia.rs

```diff
@@ -0,0 +1,40 @@
+use std::collections::HashMap;
+
+use cairo_vm::hint_processor::builtin_hint_processor::hint_utils::{get_integer_from_var_name, insert_value_into_ap};
+use cairo_vm::hint_processor::hint_processor_definition::HintReference;
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::vm_core::VirtualMachine;
+use cairo_vm::Felt252;
+
+use crate::hints::types::Preimage;
+use crate::hints::vars;
+
+pub const HEIGHT_IS_ZERO_OR_LEN_NODE_PREIMAGE_IS_TWO: &str =
+    "memory[ap] = 1 if ids.height == 0 or len(preimage[ids.node]) == 2 else 0";
+
+pub fn height_is_zero_or_len_node_preimage_is_two(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let height = get_integer_from_var_name(vars::ids::HEIGHT, vm, ids_data, ap_tracking)?;
+    let node = get_integer_from_var_name(vars::ids::NODE, vm, ids_data, ap_tracking)?;
+
+    let ap = if *height == Felt252::ZERO {
+        Felt252::ONE
+    } else {
+        let preimage: Preimage = exec_scopes.get(vars::scopes::PREIMAGE)?;
+        let preimage_value = preimage
```

Comment:
Don't we want preimage to be fully typed?

Reply:
Do you have a reason for that in mind? So far `pub type Preimage = HashMap<Felt252, Vec<Felt252>>;` has been enough.

Reply:
I need to check how feasible this is when implementing the descent map stuff so I'd say it can stay like this for now.